bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/callmonitor/CallMonitor.java

```diff
@@ -62,27 +63,44 @@ public CallMonitor(String ip, BoxHandler handler, ScheduledExecutorService sched
             } catch (InterruptedException e) {
             }
 
-            // create a new thread for listening to the FritzBox
-            CallMonitorThread thread = new CallMonitorThread();
-            thread.setName("OH-binding-" + handler.getThing().getUID().getAsString());
+            // create a new thread for listening to the FRITZ!Box
+            CallMonitorThread thread = new CallMonitorThread("OH-binding-" + handler.getThing().getUID().getAsString());
             thread.start();
             this.monitorThread = thread;
         }, 0, 2, TimeUnit.HOURS);
+        // initialize states of Call Monitor channels
+        resetChannels();
+    }
+
+    /**
+     * Reset channels.
+     */
+    public void resetChannels() {
+        handler.updateState(CHANNEL_CALL_INCOMING, UnDefType.UNDEF);
+        handler.updateState(CHANNEL_CALL_OUTGOING, UnDefType.UNDEF);
+        handler.updateState(CHANNEL_CALL_ACTIVE, UnDefType.UNDEF);
+        handler.updateState(CHANNEL_CALL_STATE, CALL_STATE_IDLE);
     }
 
     /**
      * Cancel the reconnect job.
      */
     public void dispose() {
         reconnectJob.cancel(true);
-        CallMonitorThread monitorThread = this.monitorThread;
-        if (monitorThread != null) {
-            monitorThread.interrupt();
-        }
+        stopThread();
     }
 
     public class CallMonitorThread extends Thread {
 
+        private class CallMonitorThreadExceptionHandler implements Thread.UncaughtExceptionHandler {
+            @Override
+            public void uncaughtException(@Nullable Thread thread, @Nullable Throwable throwable) {
+                logger.debug("Lost connection to FRITZ!Box because of an uncaught exception: ", throwable);
+                handler.setStatusInfo(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, String.format(
+                        "Lost connection to FRITZ!Box: %s", throwable == null ? "null" : throwable.getMessage()));
```

Comment:
Isn't this a bit harsh? My suggestion would be making `CallMonitor` implement the uncaught exception handler interface and from there logging the error and respawning the thread.

Reply:
Isn't the 2h reconnect job sufficient?

Reply:
It probably is, but I don't really see why we need to wait with reconnection when we
- know we need reconnection anyway and
- there is a trivial way to do that reconnection?
Things would be different if we needed to write a lot of code to respawn the thread, but I don't see how that would be the case here.

Reply:
I usually implement the UncaughtExceptionHandler as a lambda.

Reply:
@cpmeister Done.

@maniac103 Yes, valid points. But I think the 2h reconnect is enough. I now changed the LOG level to warn and do not set the handler to `OFFLINE` anymore.